src/libs/Violations/ViolationsUtils.ts

```diff
@@ -0,0 +1,195 @@
+                newTransactionViolations = reject(newTransactionViolations, {name: 'tagOutOfPolicy'});
+            }
+
+            // Remove 'missingTag' violation if tag is valid according to policy
+            if (hasMissingTagViolation && isTagInPolicy) {
+                newTransactionViolations = reject(newTransactionViolations, {name: 'missingTag'});
+            }
+
+            // Add 'missingTag violation' if tag is required and not set
+            if (!hasMissingTagViolation && !transaction.tag && policyRequiresTags) {
+                newTransactionViolations.push({name: 'missingTag', type: 'violation', userMessage: ''});
+            }
+        }
+
+        return {
+            onyxMethod: Onyx.METHOD.SET,
+            key: `${ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS}${transaction.transactionID}`,
+            value: newTransactionViolations,
+        };
+    },
+    /**
+     * Gets the translated message for each violation type. Necessary because `translate` throws a type error if you
+     * attempt to pass it template strings, or members of a string literal union that contains both function and string
+     * members.
+     *
+     * The issue is with the type of {@link PhraseParameters} which is defined as
+     *   `type PhraseParameters<T> = T extends (...args: infer A) => string ? A : never[];`
+     *
+     *  When a union type is passed that _may_ include a key that returns  a `string` or a key that contains a
+     * `function`, the type returns `never`.
+     *
+     *  If you have only switch cases with params, and use a default case to catch the strings, it will throw an error.
+     *
+     * @param violation
+     * @param translate
+     */
+    getViolationTranslation(violation: TransactionViolation, translate: <TKey extends TranslationPaths>(phraseKey: TKey, ...phraseParameters: PhraseParameters<Phrase<TKey>>) => string) {
+        switch (violation.name) {
+            case 'missingCategory':
+                return translate('violations.missingCategory');
+            case 'fieldRequired':
+                return translate('violations.fieldRequired');
+            case 'tagOutOfPolicy':
+                return translate('violations.tagOutOfPolicy', {tagName: violation.data?.tagName});
+            case 'missingTag':
+                return translate('violations.missingTag', {tagName: violation.data?.tagName});
+            case 'rter':
+                return translate('violations.rter', {
+                    brokenBankConnection: Boolean(violation.data?.brokenBankConnection),
+                    isAdmin: Boolean(violation.data?.isAdmin),
+                    email: violation.data?.email,
+                    isTransactionOlderThan7Days: Boolean(violation.data?.isTransactionOlderThan7Days),
+                    member: violation.data?.member,
+                });
+            case 'maxAge':
+                return translate('violations.maxAge', {maxAge: violation.data?.maxAge ?? 0});
```

Comment:
Above we do `Boolean(violation.data?.brokenBankConnection),`, but here we use `??` to fallback to a default value. Can we use the same pattern for all cases, eg, `violation.data?.brokenBankConnection ?? false` or `Number(violation.data?.maxAge)

Reply:
The two cases are different in a few ways:

1. Boolean values don't really need a fallback, since `undefined  === false` returns `true` in Javascript. So undefined is equal to the base case -- there's nothing to fall back from.
2. `Number(violation.data?.maxAge) will be `NaN` when `violation.data?.maxage` is `undefined` so we can't cast it in the same way. 
3. Boolean values will never be rendered into a string visible by the user, while other values will appear in the rendered output. For instance:
```ts
maxAge: ({maxAge}: ViolationsMaxAgeParams) => `Date older than ${maxAge} days`
```
would return `Date older than undefined days` once it goes through the stringification process. 

This is part of why I would much rather set these defaults in the translation file, so they are not divorced from the context of the string they are appearing in. The problem with that is that undefined then becomes an acceptable value, and we lose the type hints if someone reuses these strings elsewhere. 

None of this is ideal, but the type issue with `translate` is forcing us into some compromises.

Reply:
I can add the boolean fallback for clarity though rather than use the cast.